jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipRequestCustomizer.java

```diff
@@ -0,0 +1,410 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2018 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.server.handler.gzip;
+
+import java.nio.ByteBuffer;
+import java.util.Queue;
+import java.util.regex.Pattern;
+import java.util.zip.DataFormatException;
+import java.util.zip.Inflater;
+import java.util.zip.ZipException;
+
+import org.eclipse.jetty.http.BadMessageException;
+import org.eclipse.jetty.http.HttpField;
+import org.eclipse.jetty.http.HttpFields;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.io.ByteBufferPool;
+import org.eclipse.jetty.server.Connector;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.HttpInput;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.util.ArrayQueue;
+import org.eclipse.jetty.util.BufferUtil;
+import org.eclipse.jetty.util.annotation.Name;
+
+public class GzipRequestCustomizer implements HttpConfiguration.Customizer
+{
+    public static final String GZIP = "gzip";
+    private static final HttpField X_CE_GZIP = new HttpField("X-Content-Encoding","gzip");
+    private static final Pattern COMMA_GZIP = Pattern.compile(".*, *gzip");
```

Comment:
The sequence ` *` (space then star/glob/splat) would be zero or more spaces.

The sequence Greg is using would match for ...

```
"foo,gzip"
"bar, gzip"
"bar,        gzip"
```

but not 

```
"gzip"           <-- false in whole or part
"gzip, foo"      <-- false in whole or part
"gzip,bar"       <-- false in whole or part
"bar, foo, gzip" <-- false in whole, true as a sub-match
"bar, gzip, foo" <-- false in whole, true as a sub-match
```

Reply:
If you want to match whitespace, then it's probably better to use `\s*`, no?
Also, `.*` is greedy, so `bar, foo, gzip` should match, no?

Reply:
Yup, you're right, `.*` is greedy.
So yes, `bar, foo, gzip` would match.
Since the HTTP header values acceptable character don't allow whitespace other then "space", that pattern is more then acceptable.

What about the http value "compress, gzip, deflate" in that field, the pattern greg is using would be good for a sub-match, but not a whole match.

Reply:
The only patterns we can match are those that terminate with gzip, as it has to be the outermost (last) encoding applied in order for us to decode it.  Thus it is matching anything + OWS + "gzip", but as @joakime says, the only WS allows is space, so I think the pattern is correct.

Hmmm but maybe it should be `".*, *gzip$"`.... checking....

Reply:
A final $ is not required because we use the `Matcher.matches()` method, which applies only to the entire string offered.